rmf_site_editor/src/site/sdf.rs

```diff
@@ -29,21 +29,56 @@ use rmf_site_format::{
     ModelMarker, NameInSite, Pose, Rotation, Scale, WorkcellCollisionMarker, WorkcellVisualMarker,
 };
 
-// TODO(luca) reduce chances for panic and do proper error handling here
+// TODO(luca) cleanup this, there are many ways models are referenced and have to be resolved in
+// SDF between local, fuel and cached paths so the logic becomes quite complicated.
 fn compute_model_source(path: &str, uri: &str) -> AssetSource {
-    let binding = path.strip_prefix("search://").unwrap();
-    if let Some(stripped) = uri.strip_prefix("model://") {
-        // Get the org name from context, model name from this and combine
-        let org_name = binding.split("/").next().unwrap();
-        let path = org_name.to_owned() + "/" + stripped;
-        AssetSource::Remote(path)
-    } else if let Some(path_idx) = binding.rfind("/") {
-        // It's a path relative to this model, remove file and append uri
-        let (model_path, _model_name) = binding.split_at(path_idx);
-        AssetSource::Remote(model_path.to_owned() + "/" + uri)
-    } else {
-        AssetSource::Remote("".into())
+    let mut asset_source = AssetSource::from(path);
+    match asset_source {
+        AssetSource::Remote(ref mut p) | AssetSource::Search(ref mut p) => {
+            let binding = p.clone();
+            *p = if let Some(stripped) = uri.strip_prefix("model://") {
+                // Get the org name from context, model name from this and combine
+                let org_name = binding.split("/").next().unwrap();
```

Comment:
I'm concerned about using `.unwrap()` here. Couldn't a user crash this by inputting some bad text? What if we do something like `.unwrap_or("")` and allow it to just fail to find the asset later in the pipeline?

Reply:
For both this and the similar issue, changed it to return empty string but logging an error to denote exactly what went wrong, instead of doing it silently. It should be a bit more helpful in diagnosing issues https://github.com/open-rmf/rmf_site/pull/149/commits/91a8256eee8a8ff6ffffade21b518a20c7df31f1